### PR TITLE
Setup Integration Tests

### PR DESCRIPTION
**Pull request by ulisesrmzroche**
This is still rough, but it sets up the first integration test. Uses latest versions of ember and konacha.

Screen Shot 2013-04-08 at 5 34 02 PM

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ end
 gem 'active_model_serializers', github: 'rails-api/active_model_serializers'
 gem 'jquery-rails'
 gem 'ember-rails', github: 'emberjs/ember-rails'
+gem "ember-source", "1.0.0.rc2"
```
Comment (ulisesrmzroche) [inline Gemfile:19]: Updates ember.js to rc2

**Comment by ulisesrmzroche** [comment]
Also, I couldn't figure out how to get the tests working properly with App.reset() in the afterEach hook, @dgeb.

File: Gemfile (hunk 1 above)
Comment (sly7-7) [inline Gemfile:19]: I think ember-rails should take care of it, right ?

**Comment by dgeb** [comment]
@ulisesrmzroche Thanks very much for this PR. I think this style of integration test is useful, and `chai-jquery` is a good addition.

Regarding `App.reset()` - it is working in the current test suite in the `afterEach` hook. Did you try your tests without modifying `test_helper`? The current `beforeEach` hook in `test_helper` advances the application to readiness, and then resets the application in `afterEach`. This is an approach I worked out with @stefanpenner based on his recent changes to Ember's initialization code. (Granted - I need to make sure it works with some even more recent changes to Ember).

**Comment by ulisesrmzroche** [comment]
@sly7-7, I couldn't get it to work that way. Removing Gemfile.lock and running bundle install afterwards throw this: 

Screen Shot 2013-04-09 at 11 40 47 AM

**Comment by sly7-7** [comment]
@ulisesrmzroche Oh, yeah, known issue, sorry. Seeing https://github.com/emberjs/ember-rails/issues/165 it seems that the workaround is to fix the handlebars-source version to 1.0.0.rc3

**Comment by ulisesrmzroche** [comment]
@sly7-7, so is fixing ember-source to 1.0.0.rc2. That brings in the required dependencies. They're both quite ugly though. I'm going to fix that issue too while I'm at this.

**Comment by ulisesrmzroche** [comment]
@dgeb, they still run using the deferReadiness approach, though I was looking at https://github.com/emberjs/data/issues/847, and there seems to be something going on with initialization.

**Comment by sly7-7** [comment]
@ulisesrmzroche I was talking about this dependency issue this morning, it seems like it's not resolvable from our point of view. ember-rails depends on ember-source and barber. both require handlebars-source, but it seems like since there is a 1.0.0 release of handlebars in rubygems, it's taken by barber, whereas ember-source depends explicitly on 1.0.0.rc3.

**Comment by ulisesrmzroche** [comment]
@sly7-7, bummer : / , but I'd hate to throw in temp workarounds in a documentation project. What's blocking this being fixed right now?

**Comment by sly7-7** [comment]
To be honest I don't know... When handlebars releases a real 1.0.0 on rubygems, I assume this will be resolved by its own. For now, I think one of those workaround is acceptable. @dgeb what do you think ?
@ulisesrmzroche Anyway, thanks for your work, that's a nice starting point to write more tests. I hope to have the time to help you in this task.

**Comment by dgeb** [comment]
I just made a number of updates, including the latest versions of ember, ember-data, handlebars and ember-rails. I'm going to keep ember, data and handlebars vendored for now, since the latest work with promises in ember-data is dependent on the latest ember (and RSVP).

On the downside, this PR no longer merges cleanly. I took a stab at pulling in a few pieces, but the integration tests didn't pass right away. If you want to rebase and squash these commits (removing any changes to ember and test_helper), and you're able to get your new tests passing, that would be great. Otherwise, I'll try to take this on myself when I have the time.

**Comment by ulisesrmzroche** [comment]
@dgeb, I'll take a look at them right now and see if I can rewrite them. I'll let you know if I have any questions.

**Comment by dgeb** [comment]
@ulisesrmzroche I almost hate to say it, but some new [integration testing helpers](https://github.com/emberjs/ember.js/blob/v1.0.0-rc.3/packages/ember-testing/lib/helpers.js) have been added to Ember recently, so I think it would be more appropriate to use these than to introduce `chai-jquery` into this project. Let me know if you want to modify your integration test to use these helpers - otherwise, I'll take a stab at it soon. Sorry for all the churn on this PR.

**Comment by sly7-7** [comment]
@dgeb @ulisesrmzroche I can take alook a this, and use the latest ember testing package in ordrer to do this.

**Comment by ulisesrmzroche** [comment]
Yeah, that'd be badass. I've been doing some work with the latest ember-testing but only for about a week or two. Would love to get some better examples. Are we going to go the qunit route or is it framework agnostic now?

**Comment by dgeb** [comment]
@sly7-7 Yeah, thanks in advance! Keep me posted... I will try to free up some time soon to help bring this up to date.

@ulisesrmzroche Unfortunately, mocha is the only framework supported by konacha, so qunit is not an option for this particular example. However, ember-testing is now framework agnostic. I've been able to use it with mocha with the help of https://github.com/teddyzeenny/ember-mocha-adapter

**Comment by sly7-7** [comment]
@dgeb I will take one hour or two this evening, let's see what I'm able to do with this bunch of time

**Pull request closed without merging**